src/main/scala/org/broadinstitute/dsde/workbench/leonardo/dao/google/HttpGoogleDataprocDAO.scala

```diff
@@ -65,9 +65,15 @@ class HttpGoogleDataprocDAO(appName: String,
 
     val request = dataproc.projects().regions().clusters().create(googleProject.value, defaultRegion, cluster)
 
-    retryWhen500orGoogleError(() => executeGoogleRequest(request)).map { op =>
+    retryWithRecoverWhen500orGoogleError { () =>
+      executeGoogleRequest(request)
+    } {
+      case e: GoogleJsonResponseException if e.getStatusCode == 403 &&
+        (e.getDetails.getErrors.asScala.head.getDomain.equalsIgnoreCase("usageLimits")) => throw BucketObjectAccessException.apply(clusterServiceAccount, initScript, e.getMessage)
```

Comment:
Why are we throwing a `BucketObjectAccessException` in this case? Seems like it should be a new/different exception relating to dataproc being disabled.

Reply:
Ah good catch. I agree - a different exception type would be better here.

Reply:
Also, do we want to prevent retry every time we get a 403 with  domain : usageLimit`, or should this be a little more specific?